infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -106,7 +108,11 @@
     public LogUnitServer(ServerContext serverContext) {
         this.opts = serverContext.getServerConfig();
         this.serverContext = serverContext;
-        maxCacheSize = Utils.parseLong(opts.get("--max-cache"));
+        long maxCacheSizeFromUser = Utils.parseLong(opts.get("--max-cache"));
+
+        /* Ensure that maxCacheSize is at most CACHE_HEAP_RATIO * maxHeapSize */
```

Comment:
Agree on the compiling, the cache ratio should be handed as a parameter. On the other hand, if we can prevent the user to do something wrong or something that doesn't make sense I think we should do it.
Let's replace max-cache parameters with cache ratio. If the ration is more than 1, this is stupid, we should not let anybody do this. What do you think?